### PR TITLE
egressip pkg: egressIP object handler and test added

### PR DESCRIPTION
**Pull request by elenagerman**
egressIP object handler and test added

### CODE REVIEW

**Comment by elenagerman** [comment]
@kononovn @klaskosk @trewest PTAL

File: pkg/egressip/egressip_test.go [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+	}{
+		{
+			name:              egressTestName,
+			namespaceSelector: egressTestNamespaceSelector,
+			expectedError:     "",
+		},
+		{
+			name:              egressTestName,
+			namespaceSelector: metav1.LabelSelector{},
+			expectedError:     "EgressIP 'podSelector' cannot be empty",
```
Comment (klaskosk) [inline pkg/egressip/egressip_test.go:117]:
```suggestion
			expectedError:     "EgressIP 'namespaceSelector' cannot be empty",
```

File: pkg/egressip/egressip_test.go [hunk 2]
```diff
@@ -0,0 +1,403 @@
+package egressip
```
Comment (klaskosk) [inline pkg/egressip/egressip_test.go:1]: I don't see tests for Delete and Exists, could you add them please?

Comment (elenagerman) [reply pkg/egressip/egressip_test.go:1]: done

File: pkg/egressip/egressip.go [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+	return builder
+}
+
+// WithNamespaceSelector applies namespaceSelector to the EgressIP spec definition.
+func (builder *EgressIPBuilder) WithNamespaceSelector(namespaceSelector metav1.LabelSelector) *EgressIPBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Applying podSelector %s to EgressIP %q",
```
Comment (klaskosk) [inline pkg/egressip/egressip.go:95]:
```suggestion
	glog.V(100).Infof("Applying namespaceSelector %s to EgressIP %q",
```
one more I missed last time

**Review by klaskosk — COMMENTED** [review]
Just the one minor comment, otherwise lgtm

**Review by klaskosk — DISMISSED** [review]
/lgtm

**Comment by elenagerman** [comment]
@kononovn @trewest @cdvultur @achuzhoy PTAL

**Review by klaskosk — APPROVED** [review]

**Review by cdvultur — APPROVED** [review]
/lgtm

**Pull request merged**